python_dwd/download/download.py

```diff
@@ -52,10 +52,10 @@ def _download_dwd_data(remote_file: Union[str, Path]) -> BytesIO:
     try:
         with urllib.request.urlopen(file_server) as url_request:
             zip_file = BytesIO(url_request.read())
-    except urllib.error.URLError:
-        raise urllib.error.URLError(f"Error: the stationdata {file_server} couldn't be reached.")
-    except urllib.error.HTTPError:
-        pass
+    except urllib.error.URLError as e:
+        raise e(f"Error: the stationdata {file_server} couldn't be reached.")
+    except:
+        print(file_server)
```

Comment:
After introducing baec359e through #55, we should stop using `print()` within the code base. The most sane thing here would probably be to raise the very same exception for **any kind** of exception coming from the request?

So, maybe just say `except Exception as ex` here?

Reply:
What do you think about `logging` ? 
But I am fine with avoiding print statements.

Reply:
When looking at the calling site of `_download_dwd_data()`, I see a list comprehension. So, it is probably not intended to raise an exception at all as this would cancel the downloading process even for multiple files.

Of course, this depends on our intentions here.

If the downloading process should be made graceful across multiple files where single ones may be able to fail, we might (definitively) want to put out the error message using `log.error()` and just `return None`?

When already being here: There's also `raise zipfile.BadZipFile()` which might also want to get adjusted properly?

Reply:
So planning longterm:
- The download function will be replaced by something that automatically retries the download several times (over requests.adapters.HTTPAdapter). Thus there's a chance for connection error, which should be raise as the data might really exist but is just not accessable at the moment and on the other hand a http error if this file does not exist online, for which we could just ignore the error (or rather log a warning saying that one should contact us, as something might be wrong in the process of creating a file).

Reply:
@gutzbenj: Thanks for adding additional thoughts. I didn't really mean to hijack this PR in order to start a debate on principles here coming from this nit. If you see future improvements, let's divert that into a different issue acting as a reminder for the long-term plan?

@meteoDaniel: Let's just remove the `print()` and replace it through a `raise` here? Or otherwise, replace both except bodies by appropriate `log.error()` statements?

Reply:
@amotl sure. There are also several things I have in mind but it might be changed/adjusted soon.

Reply:
I have add an exception